common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -85,7 +84,7 @@ protected Recycler() {
     }
 
     protected Recycler(int maxCapacity) {
-        this.maxCapacity = Math.max(0, maxCapacity);
+        this.maxCapacity = Math.min(0, maxCapacity);
```

Comment:
sorry this was a dumb mistake, I aimed to fix it on the other line while on the web editor